cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-media.php

```diff
@@ -146,19 +180,91 @@ public function is_media( $attachment_id ) {
 	}
 
 	/**
-	 * Get a resource type based on file.(Cloudinary v1 remove mime type in post data).
+	 * Convert media extension.
 	 *
-	 * @param \WP_Post|int $attachment_id The attachment ID or object.
+	 * @param string $filename The file to convert.
+	 *
+	 * @return string|null
+	 */
+	public function convert_media_extension( $filename ) {
+
+		// Add preferred formats in future.
+		$conversion_types = array(
```

Comment:
@DavidCramer 

Let's add a filter to this mapping.

It can be that someone might want to assign a `png` to an `ai` file, instead of a `jpg`.